core/trino-server/pom.xml

```diff
@@ -38,6 +38,17 @@
         </dependency>
     </dependencies>
 
+    <!-- Needed for provisio to resolve transitive dependencies for server assembly -->
```

Comment:
Do we actually ship artifacts from this repository? I thought it was only for testing.

Reply:
Good question. We shouldn't

Reply:
we shouldn't ship them, since those classes get loaded conditionally and need to be provided by the users.

See https://github.com/trinodb/trino/blob/b90fe09f4f28ee1f2cc6a74283b311972242d673/plugin/trino-kafka/src/main/java/io/trino/plugin/kafka/schema/confluent/ConfluentModule.java#L189-L190 and https://github.com/trinodb/trino/blob/b90fe09f4f28ee1f2cc6a74283b311972242d673/plugin/trino-kafka/src/main/java/io/trino/plugin/kafka/schema/confluent/ConfluentModule.java#L256-L257

There's tests for that - https://github.com/trinodb/trino/blob/master/testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvMultinodeConfluentKafka.java has the confluent jars on classpath while other env does not and we verify that nothing fails without those classes on classpath as long as you don't use specific functionality.

Reply:
Yes. it seems that maven 4.x is just more strict. So it doesn't import repository definition from a dependency, which is a good thing I believe. It's just needed to resolve dependencies (provided one as well) but the jar is not part of the final build anyway. All good, just more strict!

Reply:
We could probably avoid this by moving the Confluent code into a new module with `<optional>true</optional>`, so that the Kafka plugin won't have a direct dependency on it, but this is fine for now.